extensions/BugModal/lib/WebService.pm

```diff
@@ -62,6 +81,26 @@ sub rest_resources {
     ]
 }
 
+sub products {
+    my $user = Bugzilla->user;
+    my @products = @{ $user->get_enterable_products };
+
+    @products = map { { name => $_->name } } @products;
+    return {
+        products     => \@products,
+    };
+}
+
+sub components {
+    my ($self, $params) = @_;
+    my $product = Bugzilla::Product->check({ name => $params->{product_name}, cache => 1 });
+    
+    my @components = map { { name => $_->name } } @{ $product->components };
```

Comment:
Sorry, you couldn't follow the comment. `_name()` is a general function which checks the validity, and returns with only the name(property) of the object. I guess, we can approach both ways here, but as dylan suggests "`_name()` for this, which will also filter out inactive components". For products, it can be this way:
```perl
sub products {
    my $user = Bugzilla->user;
    my $products = _name($user->get_enterable_products);
    my %options;
    $options{product} = $products;
    return \%options;
}
```